frontend/src/metabase-lib/compat/compat.ts

```diff
@@ -0,0 +1,96 @@
+import * as ML from "metabase-lib";
+import type { FieldFilter } from "metabase-types/api";
+import type StructuredQuery from "metabase-lib/queries/StructuredQuery";
+import LegacyFilter from "metabase-lib/queries/structured/Filter";
+
+export const getOperatorsMap = ({
+  query,
+  column,
+  stageIndex = -1,
+}: {
+  query?: ML.Query;
+  column?: ML.ColumnWithOperators;
+  stageIndex?: number;
+}) => {
+  if (!query || !column) {
+    return {};
+  }
+
+  const operators = ML.filterableColumnOperators(column);
+  return Object.fromEntries(
+    operators.map(operator => [
+      ML.displayInfo(query, stageIndex, operator).shortName,
+      operator,
+    ]),
+  );
+};
+
+export const getMlv2FilterClause = (filter: LegacyFilter, stageIndex = -1) => {
+  const mlv2Query = filter.query().question()._getMLv2Query();
```

Comment:
Not sure if `filter.query().question()` will give you the top-level query. I think `filter.query().rootQuery().getMLv2Query()` should do the thing

Reply:
Also, I think we should be calling an MLv2 query just `query` and prefix MLv1 stuff with `legacy` (`legacyQuery`, `legacyFilter`, etc.)

Reply:
.getMLv2Query is private on query so we have to 
`const query = filter.query().rootQuery().question()._getMLv2Query();` 
😜